File: app/src/main/java/com/example/ui/Signup.java
package com.example.ui;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.util.Log;
import android.view.View;
import android.view.WindowManager;
import android.widget.Toast;

import com.example.ui.RealmDB.User;
import com.google.android.material.textfield.TextInputEditText;

import java.io.Console;

import io.realm.Realm;
import io.realm.RealmChangeListener;


//import static com.example.ui.MyApplication.app;


public class Signup extends AppCompatActivity {

    private Realm realm1;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_signup);


//       final App app= new App(new AppConfiguration.Builder("ui-eygrw").build());


//        app.loginAsync(Credentials.anonymous(), new App.Callback<io.realm.mongodb.User>() {
//            @Override
//            public void onResult(App.Result<io.realm.mongodb.User> result) {
//                if(result.isSuccess()) {
//                    //configuration=new RealmConfiguration.Builder().deleteRealmIfMigrationNeeded().build();
//                    Log.d("Suxess","Success");
//                    io.realm.mongodb.User user = app.currentUser();
//                    Long partitionValue = 123L;
//                    assert user != null;
//                    syncConfiguration = new SyncConfiguration.Builder(user, partitionValue).build();
//                    Realm.setDefaultConfiguration(syncConfiguration);
//                    Realm.getInstanceAsync(syncConfiguration, new Realm.Callback() {
//                        @Override
//                        public void onSuccess(Realm realm) {
//                            realm1=realm;
//                        }
//                    });
//                }
//                else
//                {
//                    Log.d("Suxess","UNSuccess");
//                }
//
//            }
//        });


        findViewById(R.id.sign_up).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //createUser(realm1);
            }
        });
    }

    public void createUser(Realm realm)
    {
//        realm.executeTransactionAsync(new Realm.Transaction() {
//            @Override
//            public void execute(Realm realm) {
//                User user = new User( 123L);
//                //realm.createObject(User.class);
//
//
//                if (!String.valueOf(((TextInputEditText) findViewById(R.id.nameField)).getText()).isEmpty())
//                    user.setFirstName(String.valueOf(((TextInputEditText) findViewById(R.id.nameField)).getText()));
//                else
//                    ((TextInputEditText) findViewById(R.id.nameField)).setError("Enter Your First Name");
//
//                if (!String.valueOf(((TextInputEditText) findViewById(R.id.lastname_Field)).getText()).isEmpty())
//                    user.setLastName(String.valueOf(((TextInputEditText) findViewById(R.id.lastname_Field)).getText()));
//                else
//                    ((TextInputEditText) findViewById(R.id.lastname_Field)).setError("Enter Your Last Name");
//
//                if (!String.valueOf(((TextInputEditText) findViewById(R.id.emailField)).getText()).isEmpty())
//                    user.setEmail(String.valueOf(((TextInputEditText) findViewById(R.id.emailField)).getText()));
//                else
//                    ((TextInputEditText) findViewById(R.id.emailField)).setError("Enter Your Email");
//
//                if (!String.valueOf(((TextInputEditText) findViewById(R.id.numberField)).getText()).isEmpty())
//                    user.setPhoneNumber(String.valueOf(((TextInputEditText) findViewById(R.id.numberField)).getText()));
//                else
//                    ((TextInputEditText) findViewById(R.id.numberField)).setError("Enter Your Phone Number");
//
//                if (String.valueOf(((TextInputEditText) findViewById(R.id.passfield)).getText()).equals(String.valueOf(((TextInputEditText) findViewById(R.id.conpassfield)).getText()))) {
//                    user.setPassword(String.valueOf(((TextInputEditText) findViewById(R.id.passfield)).getText()));
//
//                } else {
//                    ((TextInputEditText) findViewById(R.id.conpassfield)).setError("Password does not match");
//                    ((TextInputEditText) findViewById(R.id.conpassfield)).setText("");
////                    ((TextInputEditText) findViewById(R.id.conpassfield)).;
//
//
//
//                }
//                 realm.insertOrUpdate(user);
//            }
//
//        }, new Realm.Transaction.OnSuccess() {
//            @Override
//            public void onSuccess() {
//                Toast.makeText(Signup.this, "Successful", Toast.LENGTH_SHORT).show();
//                Log.d("sync","sucess");
//                new Handler().postDelayed(new Runnable() {
//                    @Override
//                    public void run() {
//                        finish();
//                    }
//                },1000);
//
//            }
//        }, new Realm.Transaction.OnError() {
//            @Override
//            public void onError(Throwable error) {
//                Toast.makeText(Signup.this, ""+error, Toast.LENGTH_LONG).show();
//            }
//        });
//        realm.addChangeListener(new RealmChangeListener<Realm>() {
//            @Override
//            public void onChange(Realm realm) {
//                Toast.makeText(Signup.this, "Successful"+realm.toString(), Toast.LENGTH_SHORT).show();
//            }
//        });
//

    }
    @Override
    protected void onDestroy() {
        super.onDestroy();
//        realm1.close(); // Remember to close Realm when done.
    }
}

File: app/src/main/java/com/example/ui/Signup.kt
package com.example.ui

import android.content.Intent
import android.graphics.Bitmap
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.ui.Model.UserModel
import com.example.ui.databinding.ActivitySignupBinding
import com.google.android.gms.tasks.Continuation
import com.google.android.gms.tasks.OnCompleteListener
import com.google.android.gms.tasks.Task
import com.google.android.material.textfield.TextInputEditText
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageMetadata
import com.google.firebase.storage.UploadTask
import de.hdodenhof.circleimageview.CircleImageView
import java.io.ByteArrayOutputStream
import java.io.FileNotFoundException
import java.io.IOException
import java.util.*
import java.util.concurrent.Executors
import kotlin.collections.HashMap


//import static com.example.ui.MyApplication.app;
class Signup : AppCompatActivity() {
    private lateinit var userModel: UserModel
    private lateinit var binding: ActivitySignupBinding
    private lateinit var generatedFilePath: String
    private lateinit var uploadTask: UploadTask
    private lateinit var storage: FirebaseStorage
    private var picUri: Uri? = null
    private lateinit var byteArray: ByteArray
    private val PICK_IMAGE: Int=10
    private lateinit var db: FirebaseFirestore
    private lateinit var auth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
         binding = ActivitySignupBinding.inflate(layoutInflater)
        setContentView(binding.root)
        storage = FirebaseStorage.getInstance()
        db=FirebaseFirestore.getInstance()
        userModel=UserModel()
        binding.userImage.setOnClickListener(View.OnClickListener {
             val intent=Intent()
            intent.type="image/*"
            intent.action=Intent.ACTION_PICK

            val intent2=Intent(android.provider.MediaStore.ACTION_IMAGE_CAPTURE)

            val chooserIntent=Intent.createChooser(intent, "Select Picture")
            chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, arrayOf(intent2))
            startActivityForResult(chooserIntent, PICK_IMAGE);
        })
// ...
// Initialize Firebase Auth
        auth = FirebaseAuth.getInstance()
        binding.signUp.setOnClickListener(View.OnClickListener {
            val map=HashMap<String,String>()
            val random = UUID.randomUUID().toString()
            val path = "users/$random.png"
            val fireimageRef = storage.getReference(path)
            val metadata = StorageMetadata.Builder().setCustomMetadata("caption", (binding.nameField.text.toString())).build()
            uploadTask = fireimageRef.putBytes(byteArray, metadata)
            val downloadTask=uploadTask.continueWithTask{
                task->task.result!!.storage.downloadUrl
            }

            downloadTask.addOnCompleteListener(OnCompleteListener {
                generatedFilePath=  it.result.toString()
                    }).continueWithTask{
                auth.createUserWithEmailAndPassword(binding.emailField.text.toString(), binding.passfield.text.toString()).addOnCompleteListener(this) {

                    if (it.isSuccessful) {
                        Log.d("Result_ignIn", it.result.toString())

                        val editor= getSharedPreferences("Shared", MODE_PRIVATE).edit()
                        editor.putString("UID",it.result?.user?.uid).apply()
                        userModel.UID=it.result?.user?.uid.toString()
                        userModel.Email=binding.emailField.text.toString()
                        userModel.Password=binding.passfield.text.toString()
                        userModel.UserImage=generatedFilePath
                        userModel.FirstName=binding.nameField.text.toString()
                        userModel.LastName=binding.lastnameField.text.toString()
                        userModel.PhoneNumber=binding.numberField.text.toString()
//                        map.put("UID",it.result?.user?.uid.toString())
//                        map.put("Email",binding.emailField.text.toString())
//                        map.put("Password",binding.passfield.text.toString())
//                        map.put("UserImage",generatedFilePath)
//                        map.put("FirstName",binding.nameField.text.toString())
//                        map.put("LastName",binding.lastnameField.text.toString())
//                        map.put("PhoneNumber",binding.numberField.text.toString())
                        db.collection("users").add(userModel).addOnCompleteListener(OnCompleteListener {
                            startDashboard()
                        })
                    } else {
                        // If sign in fails, display a message to the user.
                        Log.w("Result_ignIn", "createUserWithEmail:failure", it.exception)
                        Toast.makeText(baseContext, "Authentication failed.", Toast.LENGTH_SHORT).show()
                    }
                }
            }

        })
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(resultCode== RESULT_OK){
            if(requestCode==10)
            { picUri = data?.data
//                data?.extras?.get("data")
                Log.d("picUri",picUri.toString())
                var bp: Bitmap? = null
                if (picUri != null) {
                    try {
                        bp = MediaStore.Images.Media.getBitmap(this.contentResolver, picUri)
                       binding.userImage.setImageBitmap(bp)
                    } catch (e: FileNotFoundException) {
                        throw RuntimeException(e)
                    } catch (e: IOException) {
                        throw RuntimeException(e)
                    }
                }
                val stream = ByteArrayOutputStream()
                bp!!.compress(Bitmap.CompressFormat.JPEG, 80, stream)
                byteArray = stream.toByteArray()
            }
        }
    }


    private fun startDashboard() {
        startActivity(Intent(this,Dashboard::class.java))
        finish()
    }

    override fun onStart(){
        super.onStart()
//        auth.currentUser?.let {
//            startActivity(Intent(this,Dashboard::class.java))
//        }
    }
}